libgnucash/backend/xml/gnc-xml-backend.cpp

```diff
@@ -143,8 +143,13 @@ GncXmlBackend::session_begin(QofSession* session, const char* new_uri,
     xaccLogSetBaseName (m_fullpath.c_str());
     PINFO ("logpath=%s", m_fullpath.empty() ? "(null)" : m_fullpath.c_str());
 
+
     if (mode == SESSION_READ_ONLY)
+    {
+        m_lockfile.clear();
+        m_lockfd = 0;
         return; // Read-only, don't care about locks.
+    }
```

Comment:
@gjanssens I think this is correct? Currently `m_lockfd` isn't set to 0 when opening READONLY, therefore `session_end` will barf when testing it against 0.

Reply:
Ok.